From 5db90f3eea98291793289d1c1937b6cef87092df Mon Sep 17 00:00:00 2001
From: Jim Hester <james.f.hester@gmail.com>
Date: Fri, 9 Oct 2020 10:58:07 -0400
Subject: [PATCH 2/3] Add const cast to fix compilation

---
 inst/include/cpp11/r_vector.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/inst/include/cpp11/r_vector.hpp b/inst/include/cpp11/r_vector.hpp
index 1d0eb49..2a57a9d 100644
--- a/inst/include/cpp11/r_vector.hpp
+++ b/inst/include/cpp11/r_vector.hpp
@@ -588,7 +588,7 @@ r_vector<T>::proxy::proxy(SEXP data, const R_xlen_t index, T* const p, bool is_a
 template <typename T>
 inline typename r_vector<T>::proxy r_vector<T>::iterator::operator*() const {
   if (data_.is_altrep()) {
-    return proxy(data_.data(), pos_, &buf_[pos_ - block_start_], true);
+    return proxy(data_.data(), pos_, const_cast<T*>(&buf_[pos_ - block_start_]), true);
   } else {
     return proxy(data_.data(), pos_,
                  data_.data_p_ != nullptr ? &data_.data_p_[pos_] : nullptr, false);
-- 
2.28.0

